src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -646,31 +650,26 @@ public CourseAttributes updateCourseCascade(CourseAttributes.UpdateOptions updat
     }
 
     /**
-     * Permanently deletes a course in Recycle Bin by its given corresponding ID.
-     * This will also cascade the data in other databases which are related to this course.
+     * Deletes a course cascade its students, instructors, sessions, responses and comments.
+     *
+     * <p>Fails silently if no such course.
      */
     public void deleteCourseCascade(String courseId) {
-        studentsLogic.deleteStudentsForCourse(courseId);
-        instructorsLogic.deleteInstructorsForCourse(courseId);
-        feedbackSessionsLogic.deleteFeedbackSessionsForCourseCascade(courseId);
-        coursesDb.deleteCourse(courseId);
-    }
-
-    /**
-     * Permanently deletes all courses in Recycle Bin.
-     * This will also cascade the data in other databases which are related to these courses.
-     */
-    public void deleteAllCoursesCascade(List<InstructorAttributes> instructorList) {
-        Assumption.assertNotNull("Supplied parameter was null", instructorList);
+        if (getCourse(courseId) == null) {
+            return;
+        }
 
-        List<String> softDeletedCourseIdList = instructorList.stream()
-                .filter(instructor -> coursesDb.getCourse(instructor.courseId).isCourseDeleted())
-                .map(InstructorAttributes::getCourseId)
-                .collect(Collectors.toList());
+        coursesDb.deleteCourse(courseId);
```

Comment:
Should this be done last? That way, if any of the following operations fail, you can safely re-run this same delete operation. If this is done first, if any of the following operations fail, you'll end up with orphan entities.